extensions/wikidata/src/org/openrefine/wikidata/qa/WikidataConstraintFetcher.java

```diff
@@ -337,4 +342,57 @@ public boolean usableOnItems(PropertyIdValue pid) {
         }
         return results;
     }
+
+    /**
+     * Returns the list of PropertyIdValues of conflicting statements
+     *
+     * @param pid:
+     *            the property having conflicts-with constraint
+     * @return
+     */
+    public List<PropertyIdValue> getConflictsWithProperties(PropertyIdValue pid) {
+        List<Statement> statementList = getConstraintsByType(pid, CONFLICTS_WITH_CONSTRAINT_QID).collect(Collectors.toList());
+        List<Value> conflictsWithProperties = new ArrayList<>();
+        for(Statement statement : statementList){
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
+            if (specs != null) {
+                conflictsWithProperties.addAll(findValues(specs, CONFLICTS_WITH_PROPERTY_PID));
+            }
+        }
+
+        if (!conflictsWithProperties.isEmpty()) {
+            return conflictsWithProperties.stream().map(e -> e == null ? null : (PropertyIdValue) e).collect(Collectors.toList());
+        }
+        return null;
+    }
+
+    /**
+     * Returns the list of all the conflicting values
+     *
+     * @param pid:
+     *            the property having conflicts-with constraint
+     * @return
+     */
+    public List<Value> getItemWithConflicts(PropertyIdValue pid, PropertyIdValue conflictingPid) {
+        List<Statement> statementList = getConstraintsByType(pid, CONFLICTS_WITH_CONSTRAINT_QID).collect(Collectors.toList());
+        List<Value> itemList = new ArrayList<>();
+        for (Statement statement : statementList) {
+            List<SnakGroup> specs = statement.getClaim().getQualifiers();
+            boolean flag = false;
+            for(SnakGroup group : specs) {
+                for (Snak snak : group.getSnaks()) {
+                    if (snak.getValue() instanceof PropertyIdValueImpl && snak.getValue().equals(conflictingPid)){
```

Comment:
Two things: it is better to avoid referring to `PropertyIdValueImpl`, you can just use `PropertyIdValue` which is the interface. Also, this type check is actually redundant with the equality check afterwards. You should just be able to do `conflictingPid.equals(snak.getValue())` (since you know `conflictingPid` is not null and its equals method will only return true for instances of `PropertyIdValue`).

Reply:
Changing PropertyIdValueImpl to PropertyIdValue sounds good to me but removing that check will not be good I guess. Because I have used that as in Snaks we get both Property and Item so we have to update flag value only in case when we get Property as Snak. So, in the 1st if condition it will work but in 2nd else if part it will not work as expected.

Reply:
Can you give me an example of a case where `snak.getValue().equals(conflictingPid)` evaluates to `true` but `snak.getValue() instanceof PropertyIdValue` does not?

Reply:
As I mentioned, it will work for sure but in the 2nd else if condition `snak.getValue() instanceof PropertyIdValue && !snak.getValue().equals(conflictingPid)`  if we remove the check to check the instance class then there might be the case that it belongs to ItemIdValue and it will make get executed as this will be true in that case `!snak.getValue().equals(conflictingPid)`. 

As an alternative, if we really want to remove the instanceof check then one thing that comes to mind is we can place the last if condition at 2nd position. It will be better I guess?

Reply:
If `snak.getValue()` is an `ItemIdValue` then `snak.getValue().equals(conflictingPid)` will evaluate to `false` - check the `equals` implementation of `ItemIdValue` :)

Reply:
I feel we both are looking at the issue differently.

> If `snak.getValue()` is an `ItemIdValue` then `snak.getValue().equals(conflictingPid)` will evaluate to `false` - check the `equals` implementation of `ItemIdValue` :)

I do agree with your explanation, that's why I'm saying it. If it evaluates to false for other ItemIdValue the `!` at the starting `!snak.getValue().equals(conflictingPid)` will make it true and it will get executed which I don't want for now. I have tried to remove the instanceof comparison but it didn't worked as expected. If you are insisting I'll try it once again for sure.

Reply:
Could you explain what you mean by "it didn't worked as expected"? Perhaps the best thing to do in this case is to put a breakpoint on this `if` and use the debugger to find out what is happening here exactly :)

Reply:
Because I want the 2nd else if condition to execute only when the case is it is not and ItemIdValue, it is a PropertyIdValue but its value is not equal to the conflictingPid. But when I remove the instanceof check from it, the condition executes even for ItemIdValue as expected according to your explanation. 

> If `snak.getValue()` is an `ItemIdValue` then `snak.getValue().equals(conflictingPid)` will evaluate to `false` - check the `equals` implementation of `ItemIdValue` :)

But I want to it to execute for PropertyIdValue only when its value isn't equal to conflictingPid.

Reply:
> But I want to it to execute for PropertyIdValue only when its value isn't equal to conflictingPid.

Yes but that is not how you should do it: you should look at the property of the snak, not the type of its values (because there could potentially be some other parameter which would also have the same type of values). So: remove all `instanceof` checks and use `snak.getProperty()` to know what sort of parameter you are handling.

Reply:
I have removed the instanceof checks. Is it okay now?

Reply:
It is much better! Now I guess we still have the issue that you are relying on the order of the qualifiers (you expect the property first, and then the items, but there is no reason why they should necessarily appear in this order).

Reply:
I tried a lot of properties having conflicts-with constraint and observed this pattern, hence proceeded with this implementation. I won't say testes all of them but some of them. The item values don't seem to have anything that relates it to the conflicting Pid I guess. What else I should do? Any suggestions?

Reply:
I would do as follows:
- create a variable to store the conflicting pid and a list of conflicting values
- loop over all qualifier snaks (as you are doing)
  + if the property of the snak is the one that specifies the conflicting pid, then set the conflicting pid to the qualifier value
  + if the property of the snak is the one that specifies the conflicting values, then add the value to the list of conflicting values

Reply:
Hope, I correctly understand what you tried to explain. :)